cfme/tests/cloud/test_instance_power_control.py

```diff
@@ -408,3 +388,193 @@ def test_power_options_from_off(setup_provider_funcscope, provider, testing_inst
     testing_instance.wait_for_instance_state_change(desired_state=testing_instance.STATE_OFF,
                                                     timeout=1200)
     check_power_options(soft_assert, testing_instance, 'off')
+
+
+class TestInstanceRESTAPI(object):
+    """ Tests using the /api/instances collection. """
+    def verify_vm_power_state(self, vm, state):
+        vm.reload()
+        if isinstance(state, (list, tuple)):
+            return vm.power_state in state
+        else:
+            return vm.power_state == state
+
+    def verify_action_result(self, rest_api, assert_success=True):
+        assert rest_api.response.status_code == 200
+        response = rest_api.response.json()
+        if 'results' in response:
+            response = response['results'][0]
+        message = response['message']
+        success = response['success']
+        if assert_success:
+            assert success
+        return success, message
+
+    @pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
+    @pytest.mark.parametrize("from_detail", [True, False], ids=["from_detail", "from_collection"])
+    def test_stop(self, setup_provider_funcscope, provider, testing_instance, verify_vm_running,
+            soft_assert, rest_api, from_detail):
+        """ Tests instance stop
+
+        Metadata:
+            test_flag: power_control, provision, rest
+        """
+        testing_instance.wait_for_instance_state_change(desired_state=testing_instance.STATE_ON)
+        vm = testing_instance.get_vm_via_rest()
+        if from_detail:
+            vm.action.stop()
+        else:
+            rest_api.collections.instances.action.stop(vm)
+        self.verify_action_result(rest_api)
+        wait_for(
+            lambda: provider.mgmt.is_vm_stopped(testing_instance.name),
+            num_sec=1200,
+            delay=20,
+            message="mgmt system check - instance stopped")
+        soft_assert(not self.verify_vm_power_state(vm, testing_instance.STATE_ON),
```

Comment:
OPTIONAL: this also looks redundant since wait_for will raise exception if vm isn't stopped for the given time.